download-github-workflow-artifact/download.js

```diff
@@ -0,0 +1,57 @@
+let fs = require('fs');
+
+module.exports = async ({github, context}, workflow_run_id, artifact_name_regex, artifact_download_dir, max_retry = 15) => {
+    const backoffDelay = (retryCount) => new Promise(resolve => setTimeout(resolve, 1000 * retryCount));
```

Comment:
Wait, doesn't the documentation state that the backoff is supposed to be exponential? If we're worried about not waiting too long at first, we can change the multiplier to 100 (from 1000).
```suggestion
    const backoffDelay = (retryCount) => new Promise(resolve => setTimeout(resolve, 100 * (1 << retryCount)));
```

Reply:
Fancy, is the suggested way faster?

Reply:
AFAIK it is pretty common to use exponential backoffs like this in networking systems and protocols. The advantage is that initial delays are lower which could make recovery from one-off errors faster but later delays are much longer. We should probably set the retryCount to something like 8 if we want to use exponential backoff.

Reply:
ooh I see, hmm so with a retry count of 15 and if it retryCount actually hits 15 for an artifact this would sleep for 1638 seconds (~27 min) before trying again for the final time. Which in this usecase is massive overkill.

What I could do is implement exponential backoff, but have an upper ceiling for this delay (maybe ~25s?)

```suggestion
    const backoffDelay = (retryCount) => new Promise(resolve => setTimeout(resolve, 100 * (1 << Math.min(retryCount, 8)));
```

Or something along these lines.

This would still retry up to the number the user provides for the retryCount, but the delay would get capped at a ceiling (100 * (1 << 8)) which is ~26s delay.

So the sample delay increment for an artifact that takes 10 tries would be closer to:
```
retryCount  |  Delay
    1           0.2
    2           0.4
    3           0.8
    4           1.6
    5           3.2
    6           6.4
    7          12.8
    8          25.6
    9          25.6
   10          25.6
```

Thoughts @Mandrenkov ? I want to avoid a 1min+ delay between retries

Reply:
I also typo'd in the action.yml description. If you check the PR description it is currently a linear backoff (idk why I wrote exponential in the action description) ...

Reply:
Yeah, I quite like the `min(retryCount, 8)` idea! We could also write
```javascript
100 * Math.min(300, 1 << retryCount)
```
to more explicitly cap the delay at 30s.

Reply:
Updated!